### ISSUE TITLE
Open source data files

### ISSUE THREAD

**Issue by x90nik** [altmp/altv-issues#2294]
### Description of the problem

At this time, we have no control over `bin` files in `data` folder.
Let us to control loading GTA V assets in `data` folder.
`pedmodels.bin`
`vehmodels.bin`
`vehmods.bin`
`weaponmodels.bin`

### Desired solution for the problem

No response

### Alternatives you considered

Another option for that is convert `bin` files to `json`.

### Additional context

#2295

**Comment by xxshady** [comment]
why do you need it?

**Comment by mrgharabaghi** [comment]
I want to control over the assets which load to my server from the GTA V official assets.

**Comment by xxshady** [comment]
what is the use case?

**Comment by xxshady** [comment]
are bin files even related to what is being loaded in the game? isnt it info for the server so it can validate your code without game

**Comment by mrgharabaghi** [comment]
> what is the use case?

One of them is related to GTA V latest update (version `3258`).
If I able to modify data files, I able to load the new vehicles that are added to update `3258`.

> are bin files even related to what is being loaded in the game? isnt it info for the server so it can validate your code without game

Yes. For example, if you remove `vehmodels.bin`, you can't spawn `GTA V`'s vanilla  vehicles.
```
/car adder
[Error] Failed to create vehicle: 3078201489 is not a proper model hash
```

**Comment by mrgharabaghi** [comment]
Another option for that is convert `bin` files to `json`.

**Comment by xxshady** [comment]
> One of them is related to GTA V latest update (version 3258).
If I able to modify data files, I able to load the new vehicles that are added to version 3258

if alt:V would support new gta version then you could spawn these vehicles even without .bin using createVehicle native

**Comment by xxshady** [comment]
.bin is data for server to validate what you are doing, it doesnt mean these vehicles are not available in the game

**Comment by mrgharabaghi** [comment]
> if alt:V would support new gta version then you could spawn these vehicles even without .bin using createVehicle native

1. If I spawn a vehicle with client side native, what happend to sync with other players?
2. What if alt:V doesn't support new update for 1, 3 or 6 months?

**Comment by xxshady** [comment]
1. it won't work
2. well new vehicles won'be available and changing .bin files won't help you with it

**Comment by xxshady** [comment]
and you can add any vehicle yourself via modding

**Comment by mrgharabaghi** [comment]
> 2. well new vehicles won'be available and changing .bin files won't help you with it

Are you sure about that? 100%?

**Comment by xxshady** [comment]
yes

**Comment by xxshady** [comment]
.bin files do not contain vehicle models, textures, etc.

**Comment by mrgharabaghi** [comment]
Why aren't the `bin` files open sourced yet?
Is there something security related about it that shouldn't be made public?

**Comment by xxshady** [comment]
no idea

**Comment by mrgharabaghi** [comment]
And another issue with `bin` files is that if you use `dlcWhitelist` in `server.toml` like this:
```toml
dlcWhitelist = [
    'mpbeach'
]
```
You are still able to spawn `caracara2` on server side without any error.
After creating this model, car object will added to server objects, but it doesn't appeared on client side.
`caracara2` added in `vinewood` DLC.

**Comment by xxshady** [comment]
you can create new bug issue for that if it doesnt exists already

**Comment by 15322107** [comment]
> * it won't be synced
> * well new vehicles won'be available and changing .bin files won't help you with it

2021-2022 some community/mod members provided .bin files for new gta updates and then you could only spawn the new vehicles if you used the new files (if they were not updated on the cdn), so it has something to do with spawning new vehicles except it has changed since then.

[Example 1 - c00kie provides files](https://discord.com/channels/371265202378899476/583109900024938527/922303701585129473)
[Example 2 (actual question on spawning vehicle)](https://discord.com/channels/371265202378899476/371271207384907776/968152789232537670)

Example 2 is German, so you would need to use a translator.

**Comment by xxshady** [comment]
yes vehmodels.bin allows you to create vehicle on server if model represents in vehmodels.bin

**Comment by xxshady** [comment]
but if vehicle model does not exists in the game it will only be created on server but not in game

**Comment by xxshady** [comment]
thats what happens here: https://github.com/altmp/altv-issues/issues/2294#issuecomment-2198412388

**Comment by xxshady** [comment]
and same will happen if you will add new vehicle model to vehmodels.bin but dont add it to the game

**Comment by 15322107** [comment]
Yes, I think what he meant, or I misunderstood.

**Comment by xxshady** [comment]
so this feature request doesnt make sense to me

**Comment by xxshady** [comment]
why would you modify vehmodels.bin if you dont have model in game?

**Comment by xxshady** [comment]
it should be handled by alt:V for you

**Comment by 15322107** [comment]
The only reason could be so he can update it himself instead of waiting for the official files, but yeah, it is not 'public' for a reason, I guess.

**Comment by xxshady** [comment]
> The only reason could be so he can update it himself

what do you mean by that? he he would adapt alt:V to new game version himself?

**Comment by 15322107** [comment]
> > The only reason could be so he can update it himself
> 
> what do you mean by that? how he would adapt alt:V to new game version himself?

No, if alt:V has been updated to the new GTA version and the date files have not been updated

**Comment by xxshady** [comment]
they are always updated after gta update

**Comment by xxshady** [comment]
you don't need to do it yourself

**Comment by xxshady** [comment]
changing format to something more open like json may can also cause false bug reports, questions, misunderstandings as it is happening now xd

**Comment by mrgharabaghi** [comment]
This is not a question. https://github.com/altmp/altv-issues/issues/2294#event-13342554131
Like other docs, we can provide essential information about data files, and convert them to `json` for people.

**Comment by xxshady** [comment]
i agree about docs, still don't see any usecase for human readable format
